nilearn/image/image.py

```diff
@@ -1285,6 +1285,12 @@ def clean_img(
     # Put results back into Niimg-like object
     if mask_img is not None:
         imgs_ = masking.unmask(data, mask_img)
+    elif 'sample_mask' in clean_kwargs :
+        shape_list = list(imgs_.shape[:3])
+        shape_list.append(clean_kwargs['sample_mask'].shape[0])
+        imgs_ = new_img_like(
+            imgs_, data.T.reshape(tuple(shape_list)), copy_header=True
+        ) 
     else:
         imgs_ = new_img_like(
             imgs_, data.T.reshape(imgs_.shape), copy_header=True
```

Comment:
You can simplify this a bit ~~and I think it needs to be an if statement otherwise it won't run if mask_img is not equal to None~~. I suggest replacing L1288-L1297 with:

```python
    elif 'sample_mask' in clean_kwargs:
        sample_shape = imgs_.shape[:3] + clean_kwargs['sample_mask'].shape
    else:
        sample_shape = imgs_.shape

    imgs_ = new_img_like(
        imgs_, data.T.reshape(sample_shape), copy_header=True
    )
```
**edit** elif makes sense I spoke too soon :)